### PR TITLE
Espresso tests

### PR DESCRIPTION
**Pull request by coreylevine**
Merge espresso tests to master

### CODE REVIEW

**Comment by coreylevine** [comment]
Espresso tests pass locally:
favouritetest
hinttest
hinttogglecheckbox
shuffletest

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/HintTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+//import static android.support.test.espresso.assertion.ViewAssertions.matches;
+//import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
+//import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+//import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+//import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+//import static android.support.test.espresso.matcher.ViewMatchers.withId;
+//import static android.support.test.espresso.matcher.ViewMatchers.withParent;
+//import static android.support.test.espresso.matcher.ViewMatchers.withText;
+//import static org.hamcrest.Matchers.allOf;
+//import static org.hamcrest.Matchers.is;
```
Comment (j-suguma) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/HintTest.java]: android imports should be above third party and java/javax imports

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/FavouriteTest.java [hunk 2]
```diff
@@ -1,114 +1,115 @@
...
+//import static android.support.test.espresso.Espresso.onView;
+//import static android.support.test.espresso.action.ViewActions.click;
+//import static android.support.test.espresso.action.ViewActions.scrollTo;
+//import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
+//import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+//import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+//import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+//import static android.support.test.espresso.matcher.ViewMatchers.withId;
+//import static android.support.test.espresso.matcher.ViewMatchers.withParent;
+//import static android.support.test.espresso.matcher.ViewMatchers.withText;
```
Comment (j-suguma) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/FavouriteTest.java]: android imports should be listed first, followed by third party, then java. Also group the hamcrest imports together

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/HintToggleCheckboxTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+//import static android.support.test.espresso.Espresso.onView;
+//import static android.support.test.espresso.action.ViewActions.click;
+//import static android.support.test.espresso.action.ViewActions.scrollTo;
+//import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+//import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+//import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+//import static android.support.test.espresso.matcher.ViewMatchers.withId;
+//import static android.support.test.espresso.matcher.ViewMatchers.withText;
+//import static org.hamcrest.Matchers.allOf;
+//import static org.hamcrest.Matchers.is;
```
Comment (j-suguma) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/HintToggleCheckboxTest.java]: fix import ordering

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/shuffleTest.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+//import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
+//import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+//import static android.support.test.espresso.matcher.ViewMatchers.isFocusable;
+//import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+//import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+//import static android.support.test.espresso.matcher.ViewMatchers.withId;
+//import static android.support.test.espresso.matcher.ViewMatchers.withParent;
+//import static android.support.test.espresso.matcher.ViewMatchers.withText;
+//import static org.hamcrest.Matchers.allOf;
+//import static org.hamcrest.Matchers.is;
```
Comment (j-suguma) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/shuffleTest.java]: import ordering

**Comment by j-suguma** [comment]
CI build for branch espresso4lyfe:
image

tests:
image

both build and test pass so okay to merge for that requirement

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/HintTest.java (hunk 1 above)
Comment (coreylevine) [reply app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/HintTest.java]: I will make those changes now

**Comment by coreylevine** [comment]
@j-suguma I fixed all the import ordering. Please have a look at the code again (and approve the PR :) )

**Comment by j-suguma** [comment]
@coreylevine just the HintToggleCheckboxTest.java the ordering still isnt right, android is underneath

**Comment by coreylevine** [comment]
@j-suguma check again please! Just fixed it!

**Comment by j-suguma** [comment]
Great you fixed it!

**Pull request merged**